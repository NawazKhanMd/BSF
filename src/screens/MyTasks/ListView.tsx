/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React, { useEffect, useState } from "react";
import { View, Text, FlatList, StyleSheet } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import MyButton from "../../components/Button";
import Spacer from "../../components/Spacer";
import { removeTask } from "../../store/actions";
import I18n from "../../i18n";
import { ListViewProps } from "../../utils/types";

const ListView = (props: ListViewProps) => {
  const { current, handleRemove } = props;
  const [showDisclaimer, setDisclaimer] = useState(-1);

  const handleDisclaimer = (index: number) => {
    setDisclaimer(index);
    const timer = setTimeout(() => {
      setDisclaimer(-1);
      clearTimeout(timer);
    }, 2000);
  };

  return (
    <View style={styles.listView}>
      <FlatList
        data={current}
        renderItem={({ item, index }) => (
          <>
            <MyButton
              title={item}
              height={50}
              bgColor={"#6454B6"}
              width={"100%"}
              onLongPress={() => handleRemove(index)}
              onPress={() => handleDisclaimer(index)}
            />
            {showDisclaimer === index && (
              <Text style={styles.disclaimer}>
                {I18n.t("deleteDiscalimer")}
              </Text>
            )}
            <Spacer />
          </>
        )}
      />
      <Spacer />
    </View>
  );
};

const styles = StyleSheet.create({
  listView: { flex: 1 },
  disclaimer: { textAlign: "center" },
});

export default ListView;
