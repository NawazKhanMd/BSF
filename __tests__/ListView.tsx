/**
 * @format
 */

import "react-native";
import React from "react";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import ListView from "../src/screens/MyTasks/ListView";
import MyButton from "../src/components/Button";

const handleRemove = jest.fn();

const i18n = jest.fn();
jest.mock("../src/i18n/index.ts", () => {
  return { t: () => {} };
});

it("renders correctly", () => {
  const props = { current: ["Some Task"], handleRemove };
  const testRenderer = renderer.create(<ListView {...props} />);
  const testInstance = testRenderer.root;
  expect(testInstance.findByType(MyButton).props.title).toBe("Some Task");
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
