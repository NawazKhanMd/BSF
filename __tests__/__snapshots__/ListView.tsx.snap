// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        "Some Task",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#6454B6",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 17,
                  "fontWeight": "bold",
                }
              }
            >
              Some Task
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "height": 10,
      }
    }
  />
</View>
`;
